bigdata-core/bigdata-sails/src/java/com/bigdata/rdf/sail/webapp/BigdataServlet.java

```diff
@@ -283,7 +286,7 @@ static public IIndexManager getIndexManager(final ServletContext servletContext)
             context.addTask(task, ft);
             
             // Await Future.
-            ft.get();
+            ft.get(BigdataRDFContext.getQueryTimeout(task.req, context.getConfig().queryTimeout), TimeUnit.MILLISECONDS);
```

Comment:
I prefer this

            // Await Future.
            final long timeoutMillis = BigdataRDFContext.getQueryTimeout(task.req, context.getConfig().queryTimeout);
            
            ft.get(timeoutMillis, TimeUnit.MILLISECONDS);